e2e/cypress/integration/interactive_menu/basic_options_spec.js

```diff
@@ -154,4 +154,85 @@ describe('MM-15887 Interactive menus - basic options', () => {
             });
         });
     });
+
+    it('should truncate properly the selected long basic option', () => {
+        const withLongBasicOption = [
+            {text: 'Option 0 - This is with very long option', value: 'option0'},
+            ...options,
+        ];
+        const basicOptions = getMessageMenusPayload({options: withLongBasicOption});
+
+        // # Post an incoming webhook for interactive menu with basic options and verify the post
+        cy.postIncomingWebhook({url: incomingWebhook.url, data: basicOptions}).then(() => {
+            verifyLastPost();
+        });
+    });
+
+    it('should truncate properly the selected long username option', () => {
+        const userOptions = getMessageMenusPayload({dataSource: 'users'});
+
+        // # Post an incoming webhook for interactive menu with user options and verify the post
+        cy.postIncomingWebhook({url: incomingWebhook.url, data: userOptions}).then(() => {
+            verifyLastPost();
+        });
+    });
+
+    it('should truncate properly the selected long channel display name option', () => {
+        const channelOptions = getMessageMenusPayload({dataSource: 'channels'});
+
+        cy.getCurrentTeamId().then((teamId) => {
+            // # Create channel with long display name
+            cy.apiCreateChannel(teamId, 'test-channel', `AAAA Very Long Display Name of a Channel ${Date.now()}`).then(() => {
+                // # Post an incoming webhook for interactive menu with channel options and verify the post
+                cy.postIncomingWebhook({url: incomingWebhook.url, data: channelOptions}).then(() => {
+                    verifyLastPost();
+                });
+            });
+        });
+    });
 });
+
+function verifyMessageAttachmentList(postId, isRhs, text) {
+    return cy.get(`#messageAttachmentList_${postId}`).within(() => {
+        cy.queryByTestId('autoCompleteSelector').should('be.visible');
+
+        if (isRhs) {
+            // * Verify that the selected option from center view matches the one in RHS
+            cy.get('.select-suggestion-container > input').should('have.value', text);
+        } else {
+            // # Select an option (long) in center view
+            cy.get('.select-suggestion-container > input').should('be.visible').click();
+            cy.get('#suggestionList').should('be.visible').children().first().click();
+        }
+
+        // * Verify exact height, width and padding of suggestion container and its input
+        cy.get('.select-suggestion-container').
+            should('be.visible').
+            and('have.css', 'height', '30px').
+            and('have.css', 'width', '220px');
+
+        cy.get('.select-suggestion-container > input').
+            and('have.css', 'height', '30px').
+            and('have.css', 'width', '220px').
+            and('have.css', 'padding-right', '30px');
```

Comment:
I'm not particularly sure how to exactly check if a text is properly truncated or not.  The approach I have here is to check CSS properties related to dimension with the assumption that if matches then no text overflow (expected).  If with overflow (due to not truncated properly), then dimension might blow out of proportion. 
Open to suggestion on how to effectively test this.

Reply:
Yeah, I'm not a fan of checking pixel amounts. Class names, sure, but exact pixels can get a bit hairy. Do any class names get tacked on to indicate the truncation? I'm not sure what else there would be to check. 

That said, if this is super important for us to check, and the tests passes then I'm fine merging it and reevaluating if it gets flaky.

Reply:
Yeah, it passed consistently but may be brittle on future UI change. Let's merge and just keep an eye on it.